internal/providers/providers.go

```diff
@@ -128,7 +128,7 @@ func NewProviderBuilder(
 
 // Implements returns true if the provider implements the given type.
 func (pb *ProviderBuilder) Implements(impl db.ProviderType) bool {
-	return slices.Contains(pb.p.Implements, impl)
+	return pb.p.CanImplement(impl)
```

Comment:
I have decided to leave this method for now. When my `ProviderFactory` PR comes along, the `ProviderBuilder` will be removed entirely. It doesn't seem like a good use of time to clean up all usages of this method.